lib/bundler/endpoint_specification.rb

```diff
@@ -91,6 +91,13 @@ def _local_specification
     end
 
     def __swap__(spec)
+      without_type = proc {|d| Gem::Dependency.new(d.name, d.requirements_list.sort) }
+      if (extra_deps = spec.runtime_dependencies.map(&without_type).-(dependencies.map(&without_type))) && extra_deps.any?
```

Comment:
Could you please explain to me what does `.-()` mean? 
Is it array difference [method](https://ruby-doc.org/core-2.3.3/Array.html#method-i-2D)?
Are there some pitfalls of plain `[] - []`?